### PR TITLE
Speed up PerImageStandardize

### PR DESCRIPTION
**Pull request by WindQAQ**
Address #593 . The implementation is based on [here](https://github.com/tensorflow/tensorflow/blob/r1.11/tensorflow/python/ops/image_ops_impl.py#L1176) (including adjusted stddev).
On single P100 GPU, this is 50x faster than `map_fn` for input shape `(128, 32, 32, 3)`

### CODE REVIEW

**Comment by goodfeli** [comment]
Thanks! The speed sounds great. Can you add a unit test that the 2 methods are equivalent?

**Comment by WindQAQ** [comment]
It's done!

File: cleverhans/compat.py [hunk 1]
```diff
@@ -81,6 +81,16 @@ def reduce_mean(input_tensor, axis=None, keepdims=None,
                          reduction_indices=reduction_indices)
 
 
+def reduce_prod(input_tensor, axis=None, keepdims=None,
+                name=None, reduction_indices=None):
+  """
+  Wrapper around the tf.reduce_prod to handle argument keep_dims
+  """
+  return reduce_function(tf.reduce_mean, input_tensor, axis=axis,
```
Comment (npapernot) [inline cleverhans/compat.py]: Shouldn't this be ` return reduce_function(tf.reduce_prod, input_tensor, axis=axis,` ?

File: cleverhans/picklable_model.py [hunk 2]
```diff
@@ -562,10 +563,22 @@ def get_params(self):
   def fprop(self, x, **kwargs):
     # TODO: before adding dataset augmentation, we didn't have to do this.
```
Comment (npapernot) [inline cleverhans/picklable_model.py]: I believe we can remove the TODO if we removed the with tf.device() call. Is this correct @goodfeli ?

File: cleverhans/compat.py (hunk 1 above)
Comment (npapernot) [reply cleverhans/compat.py]: Is the test sensitive enough if it did not break despite the function not being the right one?

Comment (WindQAQ) [reply cleverhans/compat.py]: It seems that the random generated samples will not trigger adjusted stddev, so the test pass even if this is wrong. Maybe I can come up with some dedicated cases for this?

**Comment by npapernot** [comment]
Yes, that would be nice to make sure that future changes to compat.py don't
break the changes made here. Thanks!

**Comment by npapernot** [comment]
Sorry I just realized why the tests were not failing, they were added to a different non-monitored repo. Can you move them to the `tests_tf` folder? You can find an example test in `tests_tf/test_attacks.py` for instance. Sorry again for not noticing this earlier.

**Comment by WindQAQ** [comment]
It's okay. 
I think it's better to create another PR for moving all of them because some of the files in `cleverhans/tests` have the same name with those in `tests_tf` (like test_utils_tf.py), and some of them have different testing styles (class wrapper or method). Therefore, I just put my test in the right place in this PR, and I will create another one to address #703

**Comment by npapernot** [comment]
Tests are failing due to a formatting issue
```
cleverhans/dataset.py:117: [W0612(unused-variable), maybe_download_file] Unused variable 'urljoin'
```

The last item remaining is to remove the TODO (see comment above). Once this is fixed, we can merge the PR

**Comment by WindQAQ** [comment]
Fixed :smiley:
